yt_dlp/extractor/common.py

```diff
@@ -1569,11 +1565,18 @@ def extract_video_object(e):
             extract_chapter_information(e)
 
         def traverse_json_ld(json_ld, at_top_level=True):
+            if isinstance(json_ld, dict):
+                json_ld = [json_ld]
+            if not isinstance(json_ld, list):
+                return
+
             for e in json_ld:
```

Comment:
```suggestion
            for e in variadic(json_ld):
```
accomplishes the same